### PR TITLE
build: remove gitmodules

### PR DESCRIPTION
**Pull request by turtlemoji**
We start to using normal github dependencies and remappings instead of forge libraries as gitmodules.

### CODE REVIEW

**Review by skeletor-spaceman — DISMISSED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -42,31 +46,32 @@ jobs:
...
-      - name: Install forge dependencies
-        run: forge install
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile --network-concurrency 1
```
Comment (0xGorilla) [inline .github/workflows/test.yml:62]: Wasn't it installing the package.json dependencies already? We had some...

File: package.json [hunk 2]
```diff
@@ -40,8 +39,12 @@
   "devDependencies": {
     "@commitlint/cli": "17.0.3",
     "@commitlint/config-conventional": "17.0.3",
+    "ds-test": "github:dapphub/ds-test#e282159",
+    "forge-std": "github:foundry-rs/forge-std#v1.5.6",
     "husky": ">=8",
+    "isolmate": "github:defi-wonderland/isolmate#59e1804",
     "lint-staged": ">=10",
+    "prb/test": "github:paulrberg/prb-test#a245c71",
```
Comment (gas1cent) [inline package.json:47]: `isolmate` and `prb-test` are production dependencies. I would add `ds-test` and `forge-std` as production too.

Comment (skeletor-spaceman) [reply package.json:47]: are they production tho? (is production the deployed contracts?)
if so the production contract uses those on compilation, which is dev.

Comment (gas1cent) [reply package.json:47]: @skeletor-spaceman my comment mostly inspired by the experience of importing price interfaces package in keep3r scripts and other repos. E.g. having `isolmate` as a dev dependency means the importing party will have to explicitly list it as their dependency or see `yarn install` failures.

Comment (skeletor-spaceman) [reply package.json:47]: oh then it is most definitely required. I agree. nice one!

File: .github/workflows/test.yml (hunk 1 above)
Comment (turtlemoji) [reply .github/workflows/test.yml:62]: nope, thats on the other yml action!

File: package.json (hunk 2 above)
Comment (turtlemoji) [reply package.json]: smart gas. Fixed 5bb9640d0b141d9fa3f2704afa6df28d12372e14

**Review by gas1cent — APPROVED** [review]
🐢❤️

**Review by gotzenx — APPROVED** [review]

**Review by 0xng — APPROVED** [review]

**Pull request merged**